app/Http/Controllers/TaskController.php

```diff
@@ -26,6 +27,23 @@ public function createTask(CreateTaskRequest $request){
 
         return response()->json(['message' => 'Task created with success !'],201);
     }
-
+    public function updateTask(UpdateTaskRequest $request,$id){
+        $task = Task::find($id);
+        if(!isset($task)){
+            return response()->json(['error'=>'Task not found'],404);
+        }
+        
+        $task->update([$request->validated()]);
```

Comment:
remove the brackets [] around `$request->validated()` replace it by
`$task->update($request->validated());`

Reply:
yeh thank you it's working now